pkg/networkservice/common/discoverforwarder/server.go

```diff
@@ -93,6 +93,18 @@ func (d *discoverForwarderServer) Request(ctx context.Context, request *networks
 			return nil, errors.New("no candidates found")
 		}
 
+		segments := request.Connection.GetPath().GetPathSegments()
+		datapathForwarder := ""
+		if pathIndex := int(request.Connection.GetPath().Index); len(segments) > pathIndex+1 {
+			datapathForwarder = segments[pathIndex+1].Name
```

Comment:
We can remove 
`datapathForwarder := ""`
and use
`datapathForwarder := segments[pathIndex+1].Name`